### PR TITLE
Update "user index" page to use React and GraphQL.

### PR DESCRIPTION
**Pull request by DFurnes**
#### What's this PR do?
This pull request updates Rogue's "user index" page to render via our new "modern" application shell & run searches via GraphQL (using the new `users` query introduced in DoSomething/graphql#140).

It also adds a new `app.blade.php` view that renders the `#app` container and has a little nicer "loading" placeholder than a blank page. 😇

#### How should this be reviewed?
👀

#### Any background context you want to provide?
I'm knocking these out first because they're low-hanging fruit & should make it easier to tackle the more complicated elements (like the campaign inbox). 🍒

#### Relevant tickets
[#169233808](https://www.pivotaltracker.com/story/show/169233808)

#### How to deploy
https://github.com/DoSomething/rogue/blob/master/docs/development/deployments.md

### CODE REVIEW

File: resources/assets/components/UserIndex.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+};
+
+/**
+ * The user index!
+ *
+ * @param {string} term
+ */
+const UserIndex = () => {
+  const [term, setTerm] = useState(null);
+  const inputElement = useRef(null);
```
Comment (DFurnes) [inline resources/assets/components/UserIndex.js]: Here's [documentation for the `useRef` hook](https://reactjs.org/docs/hooks-reference.html#useref) if you're interested!

File: app/Http/Controllers/Web/UsersController.php [hunk 2]
```diff
@@ -1,74 +0,0 @@
...
-        // Attempt to fetch all users.
-        $users = $this->registrar->search($query);
-
-        if ($users->count() === 0) {
-            return redirect()->route('users.index')->with('status', 'No user found!');
-        }
-
-        return redirect()->route('users.show', [$users->first()->id]);
-    }
-}
```
Comment (katiecrane) [inline app/Http/Controllers/Web/UsersController.php]: All the logic that was in here got moved to the frontend?

File: resources/assets/components/PagePlaceholder.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+
+const PagePlaceholder = () => (
+  <div className="placeholder flex-center-xy">
+    <div className="spinner" />
```
Comment (katiecrane) [inline resources/assets/components/PagePlaceholder.js]: For a future PR I'd love to see Tongue Cat incorporated here somehow 😝

File: resources/assets/components/UserIndex.js (hunk 1 above)
Comment (katiecrane) [reply resources/assets/components/UserIndex.js]: > Essentially, useRef is like a “box” that can hold a mutable value in its .current property.

a-ha! thanks for including the link 😸

File: app/Http/Controllers/Web/UsersController.php (hunk 2 above)
Comment (DFurnes) [reply app/Http/Controllers/Web/UsersController.php]: Yep!

File: resources/assets/components/PagePlaceholder.js (hunk 3 above)
Comment (DFurnes) [reply resources/assets/components/PagePlaceholder.js]: yessss 👅

**Pull request merged**